### PR TITLE
SW-5024-prep Refactor Votes pages for reusability

### PR DESCRIPTION
**Pull request by karthikbtf**
- Voting read-only and edit-views have similar data and scaffolding
- Extracted that out into a VotingWrapper
- Using a VotingProvider react context to provide the votes data as both views leverage url params
- removed the useFetchVotes hook
- TODO: implement voting edit details (Follow up)

Terraware App 2024-03-13 17-35-05

### CODE REVIEW

File: src/scenes/AcceleratorRouter/Voting/VotingProvider.tsx [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
+const VotingProvider = ({ children }: Props): JSX.Element => {
   const history = useHistory();
+  const query = useQuery();
   const dispatch = useAppDispatch();
   const snackbar = useSnackbar();
 
-  const [phaseVotes, setPhaseVotes] = useState<PhaseVotes>();
+  const pathParams = useParams<{ projectId: string }>();
+  const projectId = Number(pathParams.projectId);
+  const phase: Phase = (query.get('phase') as Phase) || 'Phase 1 - Feasibility Study'; // default to phase 1?
```
Comment (nickgraz) [inline src/scenes/AcceleratorRouter/Voting/VotingProvider.tsx:26]: That seems correct

File: src/scenes/AcceleratorRouter/Voting/index.tsx [hunk 2]
```diff
@@ -2,17 +2,22 @@ import { Redirect, Route, Switch } from 'react-router-dom';
...
 import VotingEdit from './VotingEdit';
+import VotingProvider from './VotingProvider';
+import VotingView from './VotingView';
 
 const VotingRouter = () => {
   return (
     <Switch>
       <Route exact path={APP_PATHS.ACCELERATOR_VOTING}>
-        <VotingView />
+        <VotingProvider>
```
Comment (nickgraz) [inline src/scenes/AcceleratorRouter/Voting/index.tsx:13]: Not a big deal, but I think you can wrap the whole switch in the provider. Not 100% sure though.

**Review by nickgraz — APPROVED** [review]
Nice work!

File: src/scenes/AcceleratorRouter/Voting/index.tsx (hunk 2 above)
Comment (karthikbtf) [reply src/scenes/AcceleratorRouter/Voting/index.tsx:13]: That would be ideal, let me give that a try. If it works I'll get that in. Thanks for the tip.

Comment (karthikbtf) [reply src/scenes/AcceleratorRouter/Voting/index.tsx:13]: That worked. Thanks!

**Pull request merged**